tiledb/ml/readers/pytorch_sparse.py

```diff
@@ -98,6 +99,43 @@ def __check_row_dims(self, x_row_idx: np.ndarray, y_row_idx: np.ndarray) -> None
                 "of TileDB arrays X, Y should be of equal domain extent inside the batch."
             )
 
+    def __to_csr(
+        self, array_id: str, buffer: Dict[str, np.array], offset: int
+    ) -> csr_matrix:
+        """
+        :param array_id: The matrix on which the transformation will have effect 'X' for x_array and 'Y' for y_array
+        :param buffer: The buffered slice of the matrix to be batched
+        :param offset: The starting offset of the buffered slice
+        :returns A CSR representation of the buffered slice of the matrix
+        """
+        if array_id not in ["X", "Y"]:
+            raise ValueError(
+                "You can either transform in CSR format inner arrays of 'X' or 'Y'"
+            )
+
+        # TODO: Only 2d arrays supported for now
+        array = self.x if array_id == "X" else self.y
+        array_dims = [array.schema.domain.dim(i).name for i in range(2)]
+
+        # Normalise indices for torch.sparse.Tensor We want the coords indices in every iteration to be
+        # in the range of [0, self.batch_size] so the torch.sparse.Tensors can be created batch-wise. If
+        # we do not normalise the sparse tensor is being created but with a dimension [0,
+        # max(coord_index)], which is overkill
+        row_size_norm = buffer[array_dims[0]].max() - buffer[array_dims[0]].min() + 1
+        col_size_norm = buffer[array_dims[1]].max() + 1
+        buffer_csr = csr_matrix(
+            (
+                buffer[
+                    self.x_attribute_names[0]
+                    if array_id == "X"
+                    else self.y_attribute_names[0]
+                ],
```

Comment:
I'd move all the if/else logic related to array_id in one place:
```
if array_id == "X":
   array = ...
   data = buffer[...]
else:
   array = ...
   data = buffer[...]
```

Reply:
I was trying to avoid repetition here but since you find it more readable and correct I will change